core/src/main/java/com/datastax/oss/driver/api/core/loadbalancing/LocalDcAwareLoadBalancingPolicy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.loadbalancing;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/** Load balancing policy taking into account local datacenter of the application. */
+public interface LocalDcAwareLoadBalancingPolicy extends LoadBalancingPolicy {
+
+  /** Returns the local datacenter name, if known; empty otherwise. */
+  @Nullable
+  String getLocalDatacenter();
+
+  /** Returns JSON string containing all properties that impact C* node connectivity. */
+  @NonNull
+  String getStartupConfiguration();
```

Comment:
See comment below about String vs DriverBaggage. Could hoist this up to LoadBalancingPolicy with an empty default as well.